### PR TITLE
Run on CPU + minor updates

### PR DESCRIPTION
**Pull request by mickaelseznec**
Hi Clément!

Thanks a lot for your pytorch port of FlowNet, it's been very useful to me.
I've just done some minor modifications to allow a run on CPU and add scipy and tqdm as explicit dependencies.

Also, path.py has been renamed path ([https://pypi.org/project/path.py/](https://pypi.org/project/path.py/ )), so now requirements.txt reflects that.

Si tu as des questions, n'hésite pas 😄 
Mickaël

### CODE REVIEW

File: run_inference.py [hunk 1]
```diff
@@ -71,7 +71,7 @@ def main():
     for ext in args.img_exts:
         test_files = data_dir.files('*1.{}'.format(ext))
         for file in test_files:
-            img_pair = file.parent / (file.namebase[:-1] + '2.{}'.format(ext))
+            img_pair = file.parent / (file.basename().splitext()[0][:-1] + '2.{}'.format(ext))
```
Comment (ClementPinard) [inline run_inference.py:74]: I think the right term here is "stem". see https://pathpy.readthedocs.io/en/latest/api.html#path.Path.stem (it was called namebase before)

File: main.py [hunk 2]
```diff
@@ -163,14 +163,20 @@ def main():
         network_data = None
         print("=> creating model '{}'".format(args.arch))
 
-    model = models.__dict__[args.arch](network_data).cuda()
-    model = torch.nn.DataParallel(model).cuda()
-    cudnn.benchmark = True
+    if device.type == "cuda":
```
Comment (ClementPinard) [inline main.py:166]: isn't `models.__dict__[args.arch](network_data).to(device)` enough ?

**Review by ClementPinard — CHANGES_REQUESTED** [review]
Hello,
thanks for your PR ! The style is indeed a bit outdated.

I think we can also get rid of tensorboardX and directly use the native tensorboard support (https://pytorch.org/docs/stable/tensorboard.html)

I commented your PR, but looks good to me !

Merci pour ton intérêt pour le repo :smiley:

**Comment by mickaelseznec** [comment]
You're absolutely right on both remarks, I've made the adequate modifications.

**Pull request merged**